gcloud-java-dns/src/test/java/com/google/gcloud/dns/DnsBatchTest.java

```diff
@@ -312,10 +340,211 @@ public void testGetProjectWithOptions() {
     String selector = (String) capturedOptions.getValue().get(PROJECT_FIELDS.rpcOption());
     assertTrue(selector.contains(Dns.ProjectField.QUOTA.selector()));
     assertTrue(selector.contains(Dns.ProjectField.PROJECT_ID.selector()));
-    DnsRpc.Callback<Project> capturedCallback = callback.getValue();
+    RpcBatch.Callback<Project> capturedCallback = callback.getValue();
     capturedCallback.onSuccess(PROJECT_INFO.toPb());
     assertEquals(PROJECT_INFO, batchResult.get());
   }
 
-  // todo(mderka) test submit and other methods when implemented
+  @Test
+  public void testListRecordSets() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<ResourceRecordSetsListResponse>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    batchMock.addListRecordSets(EasyMock.eq(ZONE_NAME), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<Page<RecordSet>> batchResult = dnsBatch.listRecordSets(ZONE_NAME);
+    assertEquals(0, capturedOptions.getValue().size());
+    assertNotNull(callback.getValue());
```

Comment:
I think that I am really stupid for not checking the constructor and attempting to mock that list. I added the tests.